website_sale_attribute_filter_category/views/templates.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Sergio Teruel
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <template id="products_attributes" inherit_id="website_sale.products_attributes" priority="9999">
+        <xpath expr="//form/ul" position="replace">
+                <h4>
+                    <i class="fa fa-filter" aria-hidden="true"/> <span>Filters by category</span>
+                </h4>
+                <t t-foreach="attribute_categories" t-as="attr_category">
+                    <t t-set="attributes_filtered" t-value="attributes.filtered(lambda x: x.category_id.id == attr_category[0])"/>
+                    <t t-set="display_category" t-value="any(len(a.value_ids) &gt;1 for a in attributes_filtered)"/>
```

Comment:
IMHO this logic would be better handled in the controller, so the view gets only the data it should display.